Alignment/OfflineValidation/bin/GCPtrends.cc

```diff
@@ -0,0 +1,272 @@
+  std::map<Int_t, std::map<Int_t, TH1D *>> Histos;
+  for (auto &level : Sublevel_Subdetector_Map) {
+    for (auto &variable : Index_Variable_Map) {
+      Histos[level.first][variable.first] = new TH1D(
+          "Histo_" + level.second + "_" + variable.second, "Histo_" + level.second + "_" + variable.second, 1, -1, 1);
+
+      for (auto &estimator : Index_Estimator_Map) {
+        Graphs[level.first][variable.first][estimator.first] = new TGraphErrors();
+      }
+    }
+  }
+
+  // loop over the comparisons (GCPs)
+  for (auto &path_run_map : comparison_map) {
+    TChain Events("alignTree", "alignTree");
+    Events.Add(path_run_map.first);
+
+    Int_t run_number = path_run_map.second;
+
+    std::cout << "       --> processing GCPtree file: " << path_run_map.first << std::endl;
+
+    TTreeReader Reader(&Events);
+    Reader.Restart();
+
+    TTreeReaderValue<Int_t> id = {Reader, "id"};
+    TTreeReaderValue<Int_t> badModuleQuality = {Reader, "badModuleQuality"};
+    TTreeReaderValue<Int_t> inModuleList = {Reader, "inModuleList"};
+    TTreeReaderValue<Int_t> level = {Reader, "level"};
+    TTreeReaderValue<Int_t> mid = {Reader, "mid"};
+    TTreeReaderValue<Int_t> mlevel = {Reader, "mlevel"};
+    TTreeReaderValue<Int_t> sublevel = {Reader, "sublevel"};
+    TTreeReaderValue<Float_t> x = {Reader, "x"};
+    TTreeReaderValue<Float_t> y = {Reader, "y"};
+    TTreeReaderValue<Float_t> z = {Reader, "z"};
+    TTreeReaderValue<Float_t> r = {Reader, "r"};
+    TTreeReaderValue<Float_t> phi = {Reader, "phi"};
+    TTreeReaderValue<Float_t> eta = {Reader, "eta"};
+    TTreeReaderValue<Float_t> alpha = {Reader, "alpha"};
+    TTreeReaderValue<Float_t> beta = {Reader, "beta"};
+    TTreeReaderValue<Float_t> gamma = {Reader, "gamma"};
+    TTreeReaderValue<Float_t> dx = {Reader, "dx"};
+    TTreeReaderValue<Float_t> dy = {Reader, "dy"};
+    TTreeReaderValue<Float_t> dz = {Reader, "dz"};
+    TTreeReaderValue<Float_t> dr = {Reader, "dr"};
+    TTreeReaderValue<Float_t> dphi = {Reader, "dphi"};
+    TTreeReaderValue<Float_t> dalpha = {Reader, "dalpha"};
+    TTreeReaderValue<Float_t> dbeta = {Reader, "dbeta"};
+    TTreeReaderValue<Float_t> dgamma = {Reader, "dgamma"};
+    TTreeReaderValue<Float_t> du = {Reader, "du"};
+    TTreeReaderValue<Float_t> dv = {Reader, "dv"};
+    TTreeReaderValue<Float_t> dw = {Reader, "dw"};
+    TTreeReaderValue<Float_t> da = {Reader, "da"};
+    TTreeReaderValue<Float_t> db = {Reader, "db"};
+    TTreeReaderValue<Float_t> dg = {Reader, "dg"};
+    TTreeReaderValue<Int_t> useDetId = {Reader, "useDetId"};
+    TTreeReaderValue<Int_t> detDim = {Reader, "detDim"};
+    TTreeReaderArray<Int_t> identifiers = {Reader, "identifiers"};
+
+    // loop over modules
+    while (Reader.Next()) {
+      if (!usebadmodules)
+        if (*badModuleQuality.Get())
+          continue;
+
+      Int_t sublevel_idx = *sublevel.Get();
+
+      Histos[sublevel_idx][0]->Fill(*dx.Get() * 10000);
```

Comment:
could the `10000` repeated here and elsewhere be a named constants (I guess it's a cm -> µm conversion factor)